src/Thumbhash.php

```diff
@@ -2,8 +2,101 @@
 
 namespace Thumbhash;
 
-class Thumbhash
+final class Thumbhash
 {
+
+    private $content;
+    private $width;
+    private $height;
+    private $pixels;
+    private $hash;
+    private $key;
+    private $url;
+
+    function __construct(string $file)
+    {
+        
+        $this->content = file_get_contents($file);
+        if (class_exists('Imagick')) {
+            list($this->width, $this->height, $this->pixels) = $this->extract_size_and_pixels_with_imagick($this->content);
+        } else {
+            list($this->width, $this->height, $this->pixels) = $this->extract_size_and_pixels_with_gd($this->content);
+        }
+
+        $this->hash = $this->RGBAToHash($this->width, $this->height, $this->pixels);
+        $this->key = $this->convertHashToString($this->hash); // You can store this in your database as a string
+    }
+
+
+
+    private function extract_size_and_pixels_with_gd(string $content): array
```

Comment:
Shouldn't be in this class. This code has nothing to do with thumbhash algorithm.